### PR TITLE
Mesh 1049/Allow signing keys to leave their identities

### PR DESCRIPTION
**Pull request by maxsam4**

### CODE REVIEW

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -1766,7 +1792,11 @@ impl<T: Trait> Module<T> {
     fn unlink_key_to_did(key: &AccountKey, did: IdentityId) {
         if let Some(linked_key_info) = <KeyToIdentityIds>::get(key) {
             match linked_key_info {
-                LinkedKeyInfo::Unique(..) => <KeyToIdentityIds>::remove(key),
+                LinkedKeyInfo::Unique(did_linked) => {
+                    if did_linked == did {
```
Comment (satyamakgec) [inline pallets/identity/src/lib.rs:1796]: I think we can ignore this if as in `leave_identity()` you are already checking valid signer status

Comment (maxsam4) [reply pallets/identity/src/lib.rs:1796]: `leave_identity` has this check but not some other functions.

File: pallets/runtime/common/tests/all/identity_test.rs [hunk 2]
```diff
@@ -548,6 +548,111 @@ fn frozen_signing_keys_cdd_verification_test_we() {
...
+}
+
+fn remove_signing_keys_test_with_externalities() {
+    let bob_key = AccountKey::from(AccountKeyring::Bob.public().0);
+
+    let bob_signing_key = SigningItem::new(Signatory::AccountKey(bob_key), vec![]);
+
+    let alice_did = register_keyring_account(AccountKeyring::Alice).unwrap();
+    let alice = Origin::signed(AccountKeyring::Alice.public());
+    let charlie_did = register_keyring_account(AccountKeyring::Charlie).unwrap();
```
Comment (vkomenda) [inline pallets/runtime/common/tests/all/identity_test.rs:566]: Unused variable.

File: pallets/runtime/common/tests/all/identity_test.rs [hunk 3]
```diff
@@ -548,6 +548,111 @@ fn frozen_signing_keys_cdd_verification_test_we() {
...
+    add_signing_item(alice_did, Signatory::from(bob_key));
+    add_signing_item(alice_did, Signatory::from(charlie_did));
+
+    // Check DidRecord.
+    let did_rec = Identity::did_records(alice_did);
+    assert_eq!(did_rec.signing_items, alice_signing_items);
+    assert_eq!(Identity::get_identity(&bob_key), Some(alice_did));
+
+    // Bob leaves
+    assert_ok!(Identity::leave_identity_as_key(bob,));
```
Comment (vkomenda) [inline pallets/runtime/common/tests/all/identity_test.rs:640]: Extra comma.

File: pallets/identity/src/lib.rs [hunk 4]
```diff
@@ -1976,6 +2006,22 @@ impl<T: Trait> Module<T> {
             Signatory::Identity(did) => Some(*did),
         }
     }
+
+    fn leave_identity(signer: Signatory, did: IdentityId) -> DispatchResult {
+        ensure!(Self::is_signer(did, &signer), Error::<T>::NotASigner);
+
+        if let Signatory::AccountKey(key) = signer {
+            Self::unlink_key_to_did(&key, did)
```
Comment (vkomenda) [inline pallets/identity/src/lib.rs:2014]: Can you correct the name as `unlink_key_from_did`?

**Pull request merged**